packages/blade/src/components/Dropdown/_decisions/decisions.md

```diff
@@ -0,0 +1,697 @@
+  />
+  <ActionListFooter
+    title="Footer Title"
+    description="Footer Description"
+    leading={<ActionListFooterIcon icon={DocIcon} />}
+    trailing={
+      <>
+        <Button variant="secondary">Secondary Button</Button>
+        <Button>Primary Button</Button>
+      </>
+    }
+  />
+</ActionList>
+```
+
+<details>
+<summary>Alternate Prop-based API</summary>
+
+**Props and constrained**
+
+```jsx
+<ActionList surfaceLevel={2}>
+  <ActionListHeader title="Recent Searches" icon={HistoryIcon} />
+  <ActionListSectionHeading title="Hello" />
+  <ActionListItem
+    title="Item Value"
+    desciption="Item Description"
+    leading={StarIcon} // Can be IconComponent type or Image URL
+    trailing="⌘ + Space"
+  />
+  <ActionListDivider />
+  <ActionListFooter
+    title="Footer Title"
+    description="Footer Description"
+    leading={DocIcon}
+    trailing={{
+      primary: {
+        onClick: () => {},
+        text: 'Primary Action',
+      },
+    }}
+    // OR
+    trailing={ArrowRightIcon}
+    // OR
+    trailing={{
+      primary: {
+        onClick: () => {
+          console.log('Feedback Yes Clicked');
+        },
+        icon: CheckIcon,
+      },
+      secondary: {
+        onClick: () => {
+          console.log('Feedback No Clicked');
+        },
+        icon: CrossIcon,
+      },
+    }}
+  />
+</ActionList>
+```
+
+</details>
+
+### `ActionList` Parent Component
+
+Wrapper for all ActionList compound components. Should handle the logic of selections.
+
+#### API
+
+Usage
+
+```jsx
+<ActionList>{/* Child components */}</ActionList>
+```
+
+Props
+
+```ts
+type ActionListProps = {
+  children: React.ReactNode;
+
+  /**
+   * Renders simple menu or checkboxes depending on the variant
+   *
+   * @default single
+   */
+  variant?: 'single' | 'multiple';
+
+  /** Callback that gets called when item is checked/clicked */
+  onChange?: () => void;
+
+  /**
+   * Changes the background surface level color
+   *
+   * @default 2
+   */
+  surfaceLevel: 2 | 3;
+};
+```
+
+### `ActionListItem`
+
+![](2022-11-29-17-29-01.png)
+
+#### Design
+
+- [Figma - ActionListItem / Body](https://www.figma.com/file/sAdplk2uYnI2ILnDKUxycW/Blade---Banking-Dark?node-id=13136%3A378703&t=RoEboUS3qWdStKK6-0)
+
+#### API
+
+Usage
+
+```jsx
+<ActionListItem
+  title="Accept API"
+  value="accept"
+  desciption="Accept this API"
+  leading={<ActionListItemIcon icon={StarIcon} />}
+  intent="negative"
+  trailing={<ActionListItemText>⌘ + Space + A</ActionListItemText>}
+  isDefaultSelected={false}
+  onClick={() => {
+    console.log('Item clicked');
+  }}
+/>
+```
+
+Props
+
+```ts
+type ActionListItemProps = {
+  title: string;
+
+  description: string;
+
+  /** Value that we use in JS form. Similar to `value` in `Checkbox` */
+  value: string;
+
+  /**
+   * Leading Icon or assets. Accepts JSX Component
+   *
+   * Should not be allowed when ActionList variant is multiple select
```

Comment:
should be allowed na what is the harm in that?

Reply:
In multiple select the checkbox appears in leading. Having an icon after the checkbox just pushes actual text to right which looks strange. So this was a design call.